engine/distributed_test.go

```diff
@@ -255,34 +248,54 @@ func TestDistributedAggregations(t *testing.T) {
 	}
 
 	for _, query := range queries {
+		query := query
 		t.Run(query.name, func(t *testing.T) {
+			t.Parallel()
+			localOpts := engine.Opts{
+				EngineOpts: promql.EngineOpts{
+					Timeout:              1 * time.Hour,
+					MaxSamples:           1e10,
+					EnableNegativeOffset: true,
+					EnableAtModifier:     true,
+				},
+			}
 			for _, test := range tests {
+				test := test
+				var allSeries []*mockSeries
+				remoteEngines := make([]api.RemoteEngine, 0, len(test.seriesSets)+1)
+				for _, s := range test.seriesSets {
+					remoteEngines = append(remoteEngines, engine.NewRemoteEngine(
+						localOpts,
```

Comment:
before we would have set lookback here; but now we declare the engines with default lookback right?

Reply:
You are right, good catch. Pulled data creation for this test again and put engine creation a bit futher down so we can use the new `localOpt` with the proper lookbackDelta.
Cyclomatic complexity on this test is really high.

Reply:
Yeah, I think at some point we need to clean up our testing story somehow!